src/audio/pipeline.c

```diff
@@ -616,6 +616,38 @@ static void pipeline_comp_trigger_sched_comp(struct pipeline *p,
 	list_item_append(&p->list, &ctx->pipelines);
 }
 
+/* if capture START/RELEASE trigger originated on dailess pipeline and reached
+ * inactive pipeline as it's source, then we indicate that it's blocked.
+ */
+static inline int pipeline_has_no_data(struct comp_dev *dependee,
+				       struct pipeline_walk_context *ctx,
+				       int dir)
```

Comment:
I would avoid negation in function names. It means that in order to check if pipeline has data you need to do !pipeline_has_no_data(). Can we go for ```pipeline_has_data```

Reply:
> I would avoid negation in function names. It means that in order to check if pipeline has data you need to do !pipeline_has_no_data(). Can we go for `pipeline_has_data`.

We should take a look at the use case, which is, and most probably always will be, to check if the pipeline is incapable of acquiring data for capture, for which we return -ENODATA response. With your logic, the only current usage would be `!pipeline_has_data()` which is on par in my opinion.

> I think we can make the return value `bool` as it can be only true or false.

Agree